genomesubsampler/prototypeSelection.py

```diff
@@ -114,30 +169,27 @@ def prototype_selection_exhaustive(dm, num_prototypes,
     def prototype_selection_exhaustive(dm: DistanceMatrix, num_prototypes: int,
     max_combinations_to_test: int=200000) -> List[str]:
     '''
-    if num_prototypes < 2:
-        raise ValueError(("'num_prototypes' must be >= 2, since a single "
-                          "prototype is useless."))
-    if num_prototypes >= len(dm.ids):
-        raise ValueError(("'num_prototypes' must be smaller than the number of"
-                          " elements in the distance matrix, otherwise no "
-                          "reduction is necessary."))
-
-    num_combinations = sp.special.binom(len(dm.ids), num_prototypes)
+    _validate_parameters(dm, num_prototypes, seedset)
+
+    ids = set(dm.ids) - set(seedset)
+    num_prototypes = num_prototypes - len(seedset)
+    num_combinations = sp.special.binom(len(ids), num_prototypes)
     if num_combinations >= max_combinations_to_test:
         raise RuntimeError(("Cowardly refuse to test %i combinations. Use a "
                             "heuristic implementation for instances with more "
                             "than %i combinations instead!")
                            % (num_combinations, max_combinations_to_test))
 
+    seedset = tuple(seedset)
```

Comment:
I don't see why an additional conversion from list to tuple should improve performance?